src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,480 @@
+    in_mondo_only_direct_source_ids_set: Set[RELATIONSHIP] = rels_direct_mondo_source.difference(rels_direct_source)
+
+    # Create outputs ---------------------------------------------------------------------------------------------------
+    robot_subheader_obs = deepcopy(ROBOT_SUBHEADER)
+    # todo: is this correct, or do we need to change or add something new like MONDO:excludeSubClassRelationship
+    robot_subheader_obs[0]['object_mondo_id'] = 'MONDO:excluded_subClassOf'
+    # - Add metadata to edges (labels, both source/Mondo IDs)
+    in_both_direct: List[Dict] = _add_edge_metadata(in_both_direct_source_ids_set, source_data_map)
+    in_mondo_only_direct: List[Dict] = _add_edge_metadata(in_mondo_only_direct_source_ids_set, source_data_map)
+    in_source_only_direct: List[Dict] = _add_edge_metadata(in_source_only_direct_source_ids_set, source_data_map)
+
+    # TODO: 1st: #3 & #5
+    #  - correct function to filter out any of not in prefix map
+    #   - for mondo ill have to not just pass 'MONDO' but I guess a prefix map for it so that this is standard
+    #  - Make sure I'm calling correct DB w/ correct params
+    #    calling both with same set of terms is working (idk why, mondo-ingest shouldn't have mondo IDs), and both
+    #    returning mondo IDs. For one, I would imagine I don't want to return Mondo IDs, but source IDs and then map
+    #    back to mondo
+
+    # TODO: Case 3: compare against ancestors (#3)
+    #  Case 3: SCR is direct in the source, but not at all in Mondo
+    #  - like case 5, ill have to do set operations after
+    #  - is this below correct?
+    t0 = datetime.now()
+    ancestors_mondo: Set[RELATIONSHIP] = _ancestors(
+        list(mondo_source_map.keys()), mondo_db, MONDO_PREFIX_MAP, True)
+    t1 = datetime.now()
+    rels_indirect_mondo = set()  # TODO
+    print(f'rels_indirect_mondo completed in {(t1 - t0).seconds} seconds')
+
+    # TODO: Case 5: compare against ancestors (#5)
+    #  Case 5: SCR is direct in Mondo, and not in the source at all
+    #   Implementation:
+    #   - change name of in_mondo_only to someting like in_mondo_only_including_indirect
+    #   - continue using 2 lines below(.difference(rels_source)), but change rels_source
+    #    - to this?: for each subject, query its ancestors, and construct a relationship. e.g. i pass
+    #      a subject, get back 10 objs. i make 10 triples (SUB, subClassOf, OBJ[i])
+    # # in_mondo_only_source_ids_set: Set[RELATIONSHIP] = rels_mondo_source.difference(rels_source)
+    # # in_mondo_only: List[Dict] = _add_edge_metadata(in_mondo_only_source_ids_set, source_data_map)
+    t0 = datetime.now()
+    # TODO: return: make sure it is correct. i think it should return source ids, then i conver to mondo ids after
+    ancestors_source: Set[RELATIONSHIP] = _ancestors(
+        list(source_data_map.keys()), ingest_db, owned_prefix_map, True)
+    rels_indirect_source = ancestors_source.difference(rels_direct_source)
```

Comment:
## Indirect relationships (@joeflack4)
- [ ] Ensure logic is correct
- [ ] Flag or fix any issues with sources
  - [ ] Why some n indirect rels so low?

<details><summary>Details</summary>
<p>

### `n` subClass relationships
These come from `mondo-ingest.db`.

|Source  |Cache size|Ancestors (tot)|Direct|Indirect|
|--------|----------|---------------|------|--------|
|Orphanet|4.1mb     |151,955        |25,549|126,406 |
|DOID    |2.2mb     |100,628        |15,505|85,123  |
|NCIT    |1.5mb     |66,720         |6,502 |60,218  |
|ICD10CM |96kb      |3,265          |1,120 |2,145   |
|ICD10WHO|1kb       |41             |18    |23      |
|OMIM    |160kb     |4,734          |4,734 |0       |

</p>
</details>

Reply:
Just _eyeballing_ the numbers (great analysis!) it looks.. I dare say.. Correct! All numbers makes sense to me, intuitively.